echo-telemetry/src/main/java/com/netflix/spinnaker/echo/telemetry/TelemetryEventListener.java

```diff
@@ -0,0 +1,168 @@
+      if (applicationId == null || applicationId.isEmpty()) {
+        log.debug("Application ID must be non-null and not empty");
+        return;
+      }
+
+      Map<String, Object> execution = (Map<String, Object>) event.getContent().get("execution");
+      if (execution == null || execution.isEmpty()) {
+        log.debug("Missing execution from Event content.");
+        return;
+      }
+
+      String hashedApplicationId = hash(applicationId);
+      String hashedExecutionId = hash(execution.getOrDefault("id", "").toString());
+      Execution.Type executionType =
+          Execution.Type.valueOf(
+              // TODO(ttomsu, louisjimenez): Add MPTv1 and v2 execution type detection.
+              execution.getOrDefault("type", "UNKNOWN").toString().toUpperCase());
+      Status executionStatus =
+          Status.valueOf(execution.getOrDefault("status", "UNKNOWN").toString().toUpperCase());
+
+      Map trigger = (Map) execution.getOrDefault("trigger", new HashMap());
+      Execution.Trigger.Type triggerType =
+          Execution.Trigger.Type.valueOf(
+              trigger.getOrDefault("type", "UNKNOWN").toString().toUpperCase());
+
+      List<Map> stages = (List<Map>) execution.getOrDefault("stages", new ArrayList<>());
+      List<Stage> protoStages = stages.stream().map(this::toStage).collect(Collectors.toList());
+
+      Execution executionProto =
+          Execution.newBuilder()
+              .setId(hashedExecutionId)
+              .setType(executionType)
+              .setStatus(executionStatus)
+              .setTrigger(Execution.Trigger.newBuilder().setType(triggerType))
+              .addAllStages(protoStages)
+              .build();
+
+      Application application =
+          Application.newBuilder().setId(hashedApplicationId).setExecution(executionProto).build();
+
+      SpinnakerInstance spinnakerInstance =
+          SpinnakerInstance.newBuilder()
+              .setId(telemetryConfigProps.getInstanceId())
+              .setVersion(telemetryConfigProps.getSpinnakerVersion())
+              .setApplication(application)
+              .build();
+
+      com.netflix.spinnaker.kork.proto.stats.Event loggedEvent =
+          com.netflix.spinnaker.kork.proto.stats.Event.newBuilder()
+              .setSpinnakerInstance(spinnakerInstance)
+              .build();
+
+      String content = JSON_PRINTER.print(loggedEvent);
+      telemetryService.log(new TypedJsonString(content));
+      log.debug("Telemetry sent!");
+    } catch (Exception e) {
+      log.warn("Could not send Telemetry event {}", event, e);
+    }
+  }
+
+  private Stage toStage(Map stage) {
+    return Stage.newBuilder()
+        .setType(stage.get("type").toString())
+        .setStatus(Status.valueOf(stage.getOrDefault("status", "UNKNOWN").toString().toUpperCase()))
+        .build();
+  }
+
+  private String hash(String clearText) {
+    return Hashing.sha256().hashString(clearText, StandardCharsets.UTF_8).toString();
```

Comment:
There was a discussion on the RFC about wanting to include some salt in this hash to make it hard to find trivially guessable application names in the data by pre-computing the hash.  I don't see that included here---was that suggestion ultimately rejected?

Reply:
I still intend to do that - however, the salt I would like to use (the application creation timestamp) is not available without an RPC call right now. I'd like to move it into the execution so that I could get the value without an additional RPC.

The logic behind this is that I think nearly every operation does a fetch from Front50 at some point or another, and that this data is static over the life of the application, and not easily guessable by someone on the outside. Happy to entertain other suggestions, though.

Reply:
It looks like there were some suggestions on the design doc around things to use.  I do think it's worth updating the design document to point out that the initial implementation won't salt these values given that it is something that a number of people commented on.

Reply:
This PR won't salt them, but that doesn't mean we're going to flip the switch without it.